src/apps/components/juttle-viewer.js

```diff
@@ -1,9 +1,43 @@
 import React from 'react';
 
-const JuttleViewer = (props) => {
-    return (
-        <pre className='juttle-source'>{props.bundle.program}</pre>
-    );
+import JuttleEditor from './juttle-editor';
+import { newBundle } from '../actions';
+import { LOCAL_BUNDLE_ID } from '../constants';
+import { Link } from 'react-router';
+
+class JuttleViewer extends React.Component {
+    constructor(props) {
+        super(props);
+        let isNewBundle = !props.bundle;
+        if (isNewBundle) {
+            props.dispatch(newBundle(LOCAL_BUNDLE_ID, {program: ''}, {}));
+        }
+    }
```

Comment:
Rethinking my last comments- if there is no runMode and a user clicks the code button, reroute to local and let the observer instantiate a new bundle